staging/src/k8s.io/kubectl/pkg/cmd/events/events.go

```diff
@@ -65,91 +60,109 @@ var (
 
+	ToPrinter func(outputWatchEvents, allNamespces bool) (printers.ResourcePrinter, error)
 	genericclioptions.IOStreams
 }
 
 // NewCmdEvents creates a new events command
-func NewCmdEvents(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
-	flags := NewEventsFlags(restClientGetter, streams)
+func NewCmdEvents(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	flags := NewEventsFlags(streams)
 
 	cmd := &cobra.Command{
-		Use:                   eventsUsageStr,
+		Use:                   fmt.Sprintf("events [(-o|--output=)%s] [--for TYPE/NAME] [--watch] [--output-watch-events] [--event=Normal,Warning]", strings.Join(flags.PrintFlags.AllowedFormats(), "|")),
 		DisableFlagsInUseLine: true,
 		Short:                 i18n.T("Experimental: List events"),
 		Long:                  eventsLong,
 		Example:               eventsExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			o, err := flags.ToOptions(cmd.Context(), args)
+			o, err := flags.ToOptions(f)
 			cmdutil.CheckErr(err)
-			cmdutil.CheckErr(o.Run())
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run(cmd.Context()))
 		},
 	}
 	flags.AddFlags(cmd)
+	flags.PrintFlags.AddFlags(cmd)
 	return cmd
 }
 
 // AddFlags registers flags for a cli.
-func (o *EventsFlags) AddFlags(cmd *cobra.Command) {
-	cmd.Flags().BoolVarP(&o.Watch, "watch", "w", o.Watch, "After listing the requested events, watch for more events.")
-	cmd.Flags().BoolVarP(&o.AllNamespaces, "all-namespaces", "A", o.AllNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
-	cmd.Flags().StringVar(&o.ForObject, "for", o.ForObject, "Filter events to only those pertaining to the specified resource.")
-	cmdutil.AddChunkSizeFlag(cmd, &o.ChunkSize)
+func (flags *EventsFlags) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVarP(&flags.Watch, "watch", "w", flags.Watch, "After listing the requested events, watch for more events.")
+	cmd.Flags().BoolVarP(&flags.AllNamespaces, "all-namespaces", "A", flags.AllNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().StringVar(&flags.ForObject, "for", flags.ForObject, "Filter events to only those pertaining to the specified resource.")
+	cmd.Flags().BoolVar(&flags.OutputWatchEvents, "output-watch-events", flags.OutputWatchEvents, "Output watch event objects when --watch is used.")
+	cmd.Flags().StringSliceVar(&flags.FilterTypes, "event", flags.FilterTypes, "Output only events in given types.")
+	cmdutil.AddChunkSizeFlag(cmd, &flags.ChunkSize)
 }
 
 // ToOptions converts from CLI inputs to runtime inputs.
-func (flags *EventsFlags) ToOptions(ctx context.Context, args []string) (*EventsOptions, error) {
+func (flags *EventsFlags) ToOptions(f cmdutil.Factory) (*EventsOptions, error) {
```

Comment:
This is step backwards, this command was previously using `RESTClientGetter` which is preferred over `cmdutil.Factory` can we go back?

Reply:
After investigating I recall why I changed from `RESTClientGetter` to `cmdutil.Factory`. The reason is for unit tests.
Because only `cmdutil.Factory.KubernetesClientSet()` [function](https://github.com/kubernetes/kubernetes/blob/fe3772890f650f9bcf020b43dc5a51fab0fa17f4/staging/src/k8s.io/kubectl/pkg/cmd/util/factory.go#L48) returns faked client for unit tests. Otherwise, tests tries to connect real cluster and fails.